asdf/treeutil.py

```diff
@@ -365,11 +370,11 @@ def _handle_immutable_sequence(node, json_id):
         return result
 
     def _handle_children(node, json_id):
-        if isinstance(node, dict):
+        if isinstance(node, Mapping):
```

Comment:
It's not safe to change this to match any `Mapping` or `MutableSequence`, since walk_and_modify relies on being able to construct a new instance of the container with no arguments.  For example, a tree containing an equivalency fails because `Equivalency.__init__` has a required argument:

```python
import asdf
from astropy.units import equivalencies

tree = {"equiv": equivalencies.spectral()}
asdf.treeutil.walk_and_modify(tree, lambda n: n)
```
```
TypeError: __init__() missing 1 required positional argument: 'equiv_list'
```